volplugin/handlers.go

```diff
@@ -261,7 +261,14 @@ func (dc *DaemonConfig) mount(w http.ResponseWriter, r *http.Request) {
 
 	volName := volConfig.String()
 
-	if dc.increaseMount(volName) > 1 {
+	// XXX docker issues unmount request after every mount failure so, this evens out
+	//     decreaseMount() in unmount
+	mountCount := dc.increaseMount(volName)
+	if !volConfig.Unlocked && mountCount > 1 {
```

Comment:
let's reorganize this a bit:

```go
if mountCount > 1 {
  if !volConfig.Unlocked {
    first block
    return
  }

  second block
  return
}
```

Pretty sure this behaves the same and is a little easier on the eyes IMO